bitnami/suitecrm/templates/deployment.yaml

```
@@ -1,164 +1,177 @@
-        livenessProbe:
-          httpGet:
-            path: /index.php
-            port: http
-            httpHeaders:
-            - name: Host
-              value: {{ include "host" . | quote }}
-          initialDelaySeconds: 300
-        readinessProbe:
-          httpGet:
-            path: /index.php
-            port: http
-            httpHeaders:
-            - name: Host
-              value: {{ include "host" . | quote }}
-          initialDelaySeconds: 60
-        resources:
-{{ toYaml .Values.resources | indent 10 }}
-        volumeMounts:
-        - name: suitecrm-data
-          mountPath: /bitnami/suitecrm
-          subPath: suitecrm
-{{- if .Values.metrics.enabled }}
-      - name: metrics
-        image: {{ template "suitecrm.metrics.image" . }}
-        imagePullPolicy: {{ .Values.metrics.image.pullPolicy | quote }}
-        command: [ '/bin/apache_exporter', '--scrape_uri', 'http://status.localhost:80/server-status/?auto']
-        ports:
+        - name: {{ include "suitecrm.fullname" . }}
+          image: {{ include "suitecrm.image" . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+          {{- if .Values.containerSecurityContext.enabled }}
+          securityContext:
+            runAsUser: {{ .Values.containerSecurityContext.runAsUser }}
+          {{- end }}
+          env:
+            - name: ALLOW_EMPTY_PASSWORD
+              value: {{ .Values.allowEmptyPassword | quote }}
+            - name: SUITECRM_VALIDATE_USER_IP
+              value: {{ .Values.suitecrmValidateUserIP | quote }}
+            {{- if .Values.mariadb.enabled }}
+            - name: SUITECRM_DATABASE_HOST
+              value: {{ include "suitecrm.mariadb.fullname" . }}
+            - name: SUITECRM_DATABASE_PORT_NUMBER
+              value: {{ .Values.mariadb.port | quote }}
+            - name: SUITECRM_DATABASE_NAME
+              value: {{ .Values.mariadb.auth.database | quote }}
+            - name: SUITECRM_DATABASE_USER
+              value: {{ .Values.mariadb.auth.username | quote }}
+            - name: SUITECRM_DATABASE_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "suitecrm.mariadb.fullname" . }}
+                  key: mariadb-password
+            {{- else }}
+            - name: SUITECRM_DATABASE_HOST
+              value: {{ .Values.externalDatabase.host | quote }}
+            - name: SUITECRM_DATABASE_PORT_NUMBER
+              value: {{ .Values.externalDatabase.port | quote }}
+            - name: SUITECRM_DATABASE_NAME
+              value: {{ .Values.externalDatabase.database | quote }}
+            - name: SUITECRM_DATABASE_USER
+              value: {{ .Values.externalDatabase.user | quote }}
+            - name: SUITECRM_DATABASE_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ printf "%s-%s" .Release.Name "externaldb" }}
+                  key: db-password
+            {{- end }}
```

You can simplify this a lot:
```suggestion
            - name: SUITECRM_DATABASE_HOST
              value: {{ include "suitecrm.databaseHost" . | quote }}
            - name: SUITECRM_DATABASE_PORT_NUMBER
              value: {{ include "suitecrm.databasePort" . | quote }}
            - name: SUITECRM_DATABASE_NAME
              value: {{ include "suitecrm.databaseName" . | quote }}
            - name: SUITECRM_DATABASE_USER
              value: {{ include "suitecrm.databaseUser" . | quote }}
            - name: SUITECRM_DATABASE_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: {{ include "suitecrm.databaseSecretName" . }}
                  key: {{ include "suitecrm.databasePasswordKey" . | quote }}
```
You'll have to add the functions `suitecrm.databaseHost`, `suitecrm.databasePort`, `suitecrm.databaseName`, `suitecrm.databaseUser`, `suitecrm.databaseSecretName` and `suitecrm.databasePasswordKey` to the **_helpers.tpl** (check Drupal)